f8a_worker/workers/graph_populator.py

```diff
@@ -0,0 +1,242 @@
+        return result
+
+    @classmethod
+    def construct_version_query(cls, input_json):
+        description = ''
+        try:
+            if len(input_json.get('analyses', {}).get('metadata', {}).get('details')) > 0:
+                description = input_json.get('analyses').get('metadata').get('details')[0].get('description', '')
+                description = cls.str_value_cleaner(description).replace("'", "\\'")
+        except:
+            # we pass and move forward without description
+            pass
+
+        drop_props = []
+        str_version = prp_version = drop_prop = ""
+        # Check if license and cve analyses succeeded. Then we refresh the property
+        if 'success' == input_json.get('analyses', {}).get('source_licenses', {}).get('status'):
+            drop_props.append('licenses')
+        if 'success' == input_json.get('analyses', {}).get('security_issues', {}).get('status'):
+            drop_props.append('cve_ids')
+        if len(drop_props) > 0:
+            drop_prop += "g.V().has('pecosystem', ecosystem).has('pname', pkg_name)" \
+                         ".has('version', version).properties('" + "','".join(drop_props) + "')." \
+                         "drop().iterate();" \
+
+        str_version += "ver = g.V().has('pecosystem', ecosystem).has('pname', pkg_name)" \
+                       ".has('version', version).tryNext().orElseGet{graph.addVertex('pecosystem', ecosystem," \
+                       "'pname', pkg_name, 'version', version, " \
+                       "'vertex_label', 'Version')};" \
+                       "ver.property('last_updated'," + str(time.time()) + ");"
+        # Add Description if not blank
+        if description:
+            prp_version += "ver.property('description','" + re.sub('[^A-Za-z0-9_\\\/\'":. ]', '', description) + "');"
+        # Get Code Metrics Details
+        if 'code_metrics' in input_json.get('analyses', {}):
+            count = 0
+            tot_complexity = 0.0
+            for lang in input_json.get('analyses').get('code_metrics').get('details', {}).get('languages', []):
+                if lang.get('metrics', {}).get('functions', {}).get('average_cyclomatic_complexity'):
+                    count += 1
+                    tot_complexity += lang['metrics']['functions']['average_cyclomatic_complexity']
+            cm_avg_cyclomatic_complexity = str(tot_complexity / count) if count > 0 else '-1'
+            cm_loc = str(input_json.get('analyses').get('code_metrics').get('summary', {}).get('total_lines', -1))
+            cm_num_files = str(input_json.get('analyses').get('code_metrics').get('summary', {})
+                               .get('total_files', -1))
+            prp_version += "ver.property('cm_num_files'," + cls.str_value_cleaner(cm_num_files) + ");" \
+                           "ver.property('cm_avg_cyclomatic_complexity'," + cls.str_value_cleaner(cm_avg_cyclomatic_complexity) + ");" \
```

Comment:
This line has too many symbols.

Reply:
Not following you. Line greater than 120 characters? I have pushed the changes, but I just kept it on purpose for better readability.

Reply:
Readability is really questionable and subjective. However, our goal is to enable linter. And if all new code will be linter ready we won't have so many issues to enable it if we follow this strictly.

So all we have to do is agree on line length. PEP8 dictates 79 chars.

Reply:
> So all we have to do is agree on line length. PEP8 dictates 79 chars.

79 chars is really low. I would suggest at least 100 (this is configurable in linters). We don't have old terminals :)